### PR TITLE
Adding changelog for v0.4.2

### PR DESCRIPTION
**Pull request by robscott**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
This adds a changelog for v0.4.2. By approving this PR, you approve a v0.4.2 release based on this commit.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/hold for #1034 to merge
/cc @bowei @hbagdi @jpeach @youngnick

### CODE REVIEW

**Comment by youngnick** [comment]
You know I'm in here. :)

/lgtm

**Comment by youngnick** [comment]
🤞

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +13,17 @@
 - [v0.1.0-rc2](#v010-rc2)
 - [v0.1.0-rc1](#v010-rc1)
 
+## v0.4.2
+
+API version: v1alpha2
```
Comment (howardjohn) [inline CHANGELOG.md:18]: didn't we add a bunch of stuff as well? like https://github.com/kubernetes-sigs/gateway-api/pull/1002/files

Comment (robscott) [reply CHANGELOG.md:18]: Good question, starting with 0.4.0, we've branched off to allow for patches independent of what's being merged to master: https://github.com/kubernetes-sigs/gateway-api/commits/release-0.4

**Comment by robscott** [comment]
Since there are no actual API changes here, I think it's safe to remove the hold and proceed with the release. We may still need more if image tagging doesn't work as expected on this initial attempt.

/hold cancel

**Pull request merged**